src/fairchem/core/trainers/base_trainer.py

```diff
@@ -294,6 +294,8 @@ def load_datasets(self) -> None:
                 self.train_dataset,
                 self.train_sampler,
             )
+        elif self.config.get("dataset", None) is None:
```

Comment:
elif "dataset" in self.config?

Reply:
Some of our configs have config['dataset']=None, so that the key is present but it is None.

Reply:
I think this is the offending logic:

https://github.com/FAIR-Chem/fairchem/blob/main/src/fairchem/core/trainers/base_trainer.py#L161

If we defaulted to `{}` instead of None there, we could clean this up. Thoughts?

Reply:
I cleaned up the logic to make it more obvious what's happening. Let me know if it still doesn't make sense!